zkvm/src/blockchain/state.rs

```diff
@@ -60,4 +60,29 @@ impl BlockchainState {
 
         Ok(())
     }
+
+    /// Executes a list of transactions, returning their tx IDs and tx logs.
+    pub fn execute_txlist(
+        txs: Vec<Tx>,
+        version: u64,
+        timestamp_ms: u64,
+    ) -> Result<Vec<(TxID, TxLog)>, BlockchainError> {
+        let bp_gens = BulletproofGens::new(256, 1);
```

Comment:
I think we want to avoid allocating our own generators inside the API. Our functions should instead accept generators from the caller and pass them where they're needed. (Double-check with @oleganza and/or @cathieyun.)

Reply:
that makes sense to me!